pkgs/by-name/un/unyaffs/package.nix

```diff
@@ -0,0 +1,40 @@
+{
+  fetchFromGitHub,
+  lib,
+  stdenv,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "unyaffs";
+  version = "0.9";
+
+  src = fetchFromGitHub {
+    owner = "whataday";
+    repo = "unyaffs";
+    rev = "${finalAttrs.version}";
+    hash = "sha256-FcuPaxq74gVJ6UlOhFPSMtuwUAJVV/sAxAQojhmVXCs=";
+  };
+
+  preBuild = ''
+    sed -i '54i #if defined(__linux__)' unyaffs.c
+    sed -i '55i #include <sys/sysmacros.h>' unyaffs.c
+    sed -i '56i #endif' unyaffs.c
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D unyaffs -t $out/bin
+
+    runHook postInstall
+  '';
+
```

Comment:
Also would be good to add https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook, if possible

Reply:
I tried the following:
```
{
  fetchFromGitHub,
  lib,
  stdenv,
  versionCheckHook,
}:

stdenv.mkDerivation (finalAttrs: {
  pname = "unyaffs";
  version = "0.9";

  src = fetchFromGitHub {
    owner = "whataday";
    repo = "unyaffs";
    rev = "${finalAttrs.version}";
    hash = "sha256-FcuPaxq74gVJ6UlOhFPSMtuwUAJVV/sAxAQojhmVXCs=";
  };

  preBuild = ''
    sed -i '54i #if defined(__linux__)' unyaffs.c
    sed -i '55i #include <sys/sysmacros.h>' unyaffs.c
    sed -i '56i #endif' unyaffs.c
  '';

  installPhase = ''
    runHook preInstall

    install -Dm755 unyaffs -t $out/bin

    runHook postInstall
  '';

  nativeInstallCheckInputs = [ versionCheckHook ];
  doInstallCheck = true;
  versionCheckProgramArg = "-V";

  meta = {
    description = "Tool to extract files from a YAFFS2 file system image";
    homepage = "https://github.com/whataday/unyaffs";
    license = lib.licenses.gpl2Only;
    platforms = lib.platforms.unix;
    maintainers = with lib.maintainers; [ KSJ2000 ];
    mainProgram = "unyaffs";
  };
})
```

But I got:
```
$ nix-build -E 'with import <nixpkgs> {}; callPackage ./package.nix {}' --show-trace
fetching path input 'path:/nix/store/vpzg5qjqwl76ichhjwwgmwc8v3j369q5-source'
error:
       … from call site
         at «string»:1:27:
            1| with import <nixpkgs> {}; callPackage ./package.nix {}
             |                           ^

       … while calling 'callPackageWith'
         at /nix/store/vpzg5qjqwl76ichhjwwgmwc8v3j369q5-source/lib/customisation.nix:212:35:
          211|   */
          212|   callPackageWith = autoArgs: fn: args:
             |                                   ^
          213|     let

       … while calling the 'abort' builtin
         at /nix/store/vpzg5qjqwl76ichhjwwgmwc8v3j369q5-source/lib/customisation.nix:269:13:
          268|        # which is especially relevant with allowAliases = false
          269|        else abort "lib.customisation.callPackageWith: ${error}";
             |             ^
          270|

       error: evaluation aborted with the following error message: 'lib.customisation.callPackageWith: Function called without required argument "versionCheckHook" at /home/KSJ2000/nixpkgs/pkgs/by-name/un/unyaffs/package.nix:5'
```
I tried to look at what others did, but it seems the same, yet here it doesn't work.
https://github.com/NixOS/nixpkgs/pull/351990/files#diff-dc6e0ddf2046f11255013739f2e307ce6e34471b4e0696681e2f29f90231b625R75
Does versionCheckHook not work here or am I calling it wrong?

Reply:
Could you try to update master branch of your fork, pull locally, and rebase (not merge) this branch?

Reply:
Hopefully did that right, but the error hasn't changed.

Reply:
Does it work for you now?

Reply:
It does not; not with: `nix-build -E 'with import <nixpkgs> {}; callPackage ./package.nix {}' --show-trace`
It gives the above error.
But if it works on your end, then it should be fine.

Reply:
> not with: `nix-build -E 'with import <nixpkgs> {}; callPackage ./package.nix {}' --show-trace`

This `<nixpkgs>` is your nixpkgs channel, not this checkout. You should use `nix-build -A foo` in project root instead,

Reply:
Yes, it works.